### DISCUSSION TITLE
TLS/SSL for JanusGraph Server

### DISCUSSION THREAD

**Discussion by amansharif** [JanusGraph/janusgraph#4671] [Q&A] [answered]
I'm new to JanusGraph and want to try it out for one of our projects. We're using Cassandra as the storage backend and Elasticsearch for the index backend. I need to enable TLS/SSL on the JanusGraph server to secure the connection between the client and the server with authentication. However, I couldn't find any documentation on SSL and authentication for JanusGraph. Here are my questions:

1. Is it possible to enable SSL on JanusGraph? If so, could you provide some guidance?
2. Could you also point me to resources on authentication and authorization for JanusGraph?

Thank you for you assistance.

**Comment by porunov** [comment, marked as answer]
JanusGraph implements Gremlin Server, so you configure your server to have necessary security settings enabled. You can find this documentation in the TinkerPop documentation: [Security](https://tinkerpop.apache.org/docs/current/reference/#security).
In your JanusGraph Server configuration you should enable ssl and provide the path to your keystore.
An example from documentation:
```
ssl: {
  enabled: true,
  sslEnabledProtocols: [TLSv1.2],
  keyStore: server.jks,
  keyStorePassword: changeit
}
```

**Reply by amansharif** [reply]
thanks for the pointer.
